CONTRIBUTING.md

```diff
@@ -1,69 +1,138 @@
-# Fork Status
+# Contribution guidelines
 
-## Pull Requests
+First off, thank you for considering contributing to ratatui.
 
-**All** pull requests opened on the original repository have been imported. We'll be going through any open PRs in a timely manner, starting with the **smallest bug fixes and README updates**. If you have an open PR make sure to let us know about it on our [discord](https://discord.gg/pMCEU9hNEj) as it helps to know you are still active.
+If your contribution is not straightforward, please first discuss the change you wish to make by
+creating a new issue before making the change.
 
-## Issues
+## Reporting issues
 
-We have been unsuccessful in importing all issues opened on the previous repository.
-For that reason, anyone wanting to **work on or discuss** an issue will have to follow the following workflow :
+Before reporting an issue on the [issue tracker](https://github.com/tui-rs-revival/ratatui/issues),
+please check that it has not already been reported by searching for some related keywords. Please
+also check [tui-rs issues](https://github.com/fdehau/tui-rs/issues/) and link any related issues
+found.
 
-- Recreate the issue
-- Start by referencing the **original issue**: ```Referencing issue #[<issue number>](<original issue link>)```
-- Then, paste the original issues **opening** text
+## Pull requests
 
-You can then resume the conversation by replying to this new issue you have created.
+All contributions are obviously welcome. Please include as many details as possible in your PR
+description to help the reviewer (follow the provided template). Make sure to highlight changes
+which may need additional attention or you are uncertain about. Any idea with a large scale impact
+on the crate or its users should ideally be discussed in a "Feature Request" issue beforehand.
+
+### Keep PRs small, intentional and focused
+
+Try to do one pull request per change. The time taken to review a PR grows exponential with the size
+of the change. Small focused PRs will generally be much more faster to approved and merge. Don't
+include large refactoring or formatting changes in PRs
```

Comment:
I'm not entirely sure what this means. There are, of course, many times where large refactors may be necessary. Formatting maybe not so much.

Reply:
When large refactoring (changes that don't alter the external behavior of some code) and actual changes that alter behavior, it can be difficult to understand which changes are relevant to the PR vs just cleanups. The chances of detecting an error are lower.

Evaluating a refactor is:
- is it easier to read / maintain
- does it do the same thing (i.e. behavior was fully covered by tests and tests continue to pass)

Evaluating a change is:
- does it do the things it says were changed
- does it have tests that have also changed / been added to cover the new / changed feature

re: formatting, something which happens often is that people will make some changes, but reformat everything around the area that they've read to make the change. Again, this makes it difficult to actually see the diff of what's changed and evaluate it. Instead this means that the reviewer has to read every line of code to understand it in full again.

I'll reword it a bit to make some of that more clear.